website_v2/gcloud-powershell-website-v2/www/static/app/referenceController.js

```diff
@@ -0,0 +1,23 @@
+﻿var app = angular.module('powershellSite');
+
+/** The controller that exposes route parameters to the templates. **/
+app.controller('ReferenceController', ['$scope', '$routeParams',
+    function($scope, $routeParams) {
+        var prodInfo = $scope.productInfo;
+        if (prodInfo === null) return;
+        /** We have to make sure no invalid routes were passed in **/
+        if (Object.keys($routeParams).length === 2) {
+            if (!($routeParams.product in prodInfo)
+                || !($routeParams.cmdlet in prodInfo[$routeParams.product])) {
+                    console.error("Invalid Product or Cmdlet");
```

Comment:
Again here logging to the console is good but I think a message in the UI is better